notebooks/official/tabular_workflows/tabnet_on_vertex_pipelines.ipynb

```diff
@@ -0,0 +1,1134 @@
+{
```

Comment:
<p>I'm sorry, it's still not clear to me how this works. If you return a range for a particular hyperparameter, will Vertex AI iterate within that range to find the optimal value? Is there a corresponding step size? Is the range given as min value / max value, as an array, something else?</p><p> </p><p>Could you provide a brief description of dataset_size_bucket, prediction_type, and training_budget_bucket? Is there any guidance for how to choose between "small", "medium", and "large"?</p><p> </p><p>Why is there a list of "recommended" hyperparameters? Why don't we have one big list and set values for all of them?  </p><p> </p><p>Please avoid future tense (we will set ranges -&gt; we set ranges).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/726/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='47'/>

Reply:
<p>If you return a range for a particular hyperparameter, will Vertex AI iterate within that range to find the optimal value? Is there a corresponding step size?</p><p>&gt;&gt; Yes. For continuous ranges, yes, but it's specified by Vertex AI.</p><p> </p><p> </p><p>&gt;&gt; Is the range given as min value / max value, as an array, something else?</p><p>Both, it depends on the parameter. <code>learning_rate</code>, for example, has min/max values.  <code>max_steps</code> has an array of discrete values (ex: [1000, 2000, 5000, 15000]).  </p><p> </p><p>&gt;&gt; Could you provide a brief description of dataset_size_bucket, prediction_type, and training_budget_bucket? Is there any guidance for how to choose between "small", "medium", and "large"?</p><p>Yes, there is. Updated the notebook. </p><p> </p><p>Why is there a list of "recommended" hyperparameters? Why don't we have one big list and set values for all of them?</p><p>&gt;&gt; The helper function does return one big list of hyperparameters/values. What I was trying to say is that it's optional. It's meant to be a guide of sorts to show what's available, but the ranges themselves can be changed. If a user doesn't care about hyperparameters or only cares about one of them, they can pass nothing or just the config for the specific hyperparameter for <code>study_spec_parameters_override</code>, and the pipeline will take what they passed and fill in ranges for the rest. </p><p> </p>